app/com/linkedin/drelephant/ElephantRunner.java

```diff
@@ -103,19 +106,16 @@ public Void run() {
           loadAnalyticJobGenerator();
           ElephantContext.init();
 
-          _jobQueue = new LinkedBlockingQueue<AnalyticJob>();
-
           // Initialize the metrics registries.
           MetricsController.init();
 
           logger.info("executor num is " + _executorNum);
-          if (_executorNum > 0) {
-            _service = Executors.newFixedThreadPool(_executorNum,
-                    new ThreadFactoryBuilder().setNameFormat("dr-el-executor-thread-%d").build());
-            for (int i = 0; i < _executorNum; i++) {
-              _service.submit(new ExecutorThread(_jobQueue));
-            }
+          if (_executorNum < 1) {
+            throw new RuntimeException("Must have at least 1 worker thread.");
           }
+          ThreadFactory factory = new ThreadFactoryBuilder().setNameFormat("dr-el-executor-thread-%d").build();
+          _threadPoolExecutor = new ThreadPoolExecutor(_executorNum, _executorNum, 0L, TimeUnit.MILLISECONDS,
```

Comment:
This is _exactly_ what `Executors.newFixedThreadPool(int nThreads, ThreadFactory threadFactory)` returns. Unfortunately, it returns something of the `ExecutorService` interface which doesn't expose any way to check the queue length. While obviously I could still cast to a `ThreadPoolExecutor`, I choose to use this slightly less fancy approach to avoid relying on implementation details.